packages/gas-fee-controller/package.json

```diff
@@ -67,7 +67,7 @@
     "typescript": "~4.9.5"
   },
   "peerDependencies": {
-    "@metamask/network-controller": "^18.1.2"
+    "@metamask/network-controller": "^18.0.0"
```

Comment:
I think this one should be aligned with the `dependencies` field? So if dropping here, then we'd also like to drop to `dependencies["@metamask/network-controller"] = "^18.1.0"` ?

Reply:
Hmm, are you saying that if a version of a package within the monorepo is bumped we should no longer sync all dependencies on that package?

Reply:
> Hmm, are you saying that if a version of a package within the monorepo is bumped we should no longer sync all dependencies on that package?

Just considering this case: Using `@metamask/gas-fee-controller`, it will itself use `@metamask/network-controller@^18.1.2` due to its dependency. It will accept any `@metamask/network-controller@^18.0.0`.
I have trouble coming up with why we want to enforce the direct dependency to be latest but still allow supplying older versions as constructor parameters or what it may be.

For packages that have the same package in both `dependencies` and `peerDependencies`, I'm thinking it's a case-by-case-basis: Either both fields are bumped, or neither. I can see both situations, but have trouble imagining why we want to have these disjoint.

Reply:
(Kind of separate, a lot of this messiness goes away of properly utilizing [worskpace cross-references](https://yarnpkg.com/features/workspaces#cross-references), I think)

Reply:
Originally, we linked `dependencies` and `peerDependencies` together. So if a developer updated the dependency of a package and it was also a peer dependency, then the versions needed to match. 

However, since bumping a peer dependency requires a major version bump, this resulted in many major version bumps. We decided to unlink them here to reduce the number of major versions: https://github.com/MetaMask/core/pull/3881. But it sounds like we should revisit this?

As for why we don't use `workspace:^`, this prevents developers from being able to test changes to core packages in extension and mobile locally (e.g. using `yarn link`). We replaced `workspace:^` references with static versions and added a Yarn constraint which ensured that if one workspace package has a dependency on another, the version of the dependency matches the current version of the package: https://github.com/MetaMask/core/pull/1623